torch/distributed/distributed_c10d.py

```diff
@@ -3622,8 +3640,8 @@ def barrier(group=GroupMember.WORLD, async_op=False, device_ids=None):
         group (ProcessGroup, optional): The process group to work on. If None,
             the default process group will be used.
         async_op (bool, optional): Whether this op should be an async op
-        device_ids ([int], optional): List of device/GPU ids.
-                                      Valid only for NCCL backend.
+        device_ids ([int], optional): List of device/GPU ids. Valid only
+            for NCCL backend when argument device is None.
```

Comment:
nit: what does "when argument device is None" mean? Does this correspond to a previous commit of this PR?